services/QuillLMS/client/app/bundles/Teacher/components/unit_templates/unit_template_activity_selector.tsx

```diff
@@ -0,0 +1,280 @@
+const SELECTED_TYPE = 'selected'
+
+const UnitTemplateActivitySelector = ({ parentActivities, setParentActivities, toggleParentActivity }) => {
+
+  const [activities, setActivities] = React.useState([])
+  const [selectedActivities, setSelectedActivities] = React.useState(parentActivities)
+  const [loading, setLoading] = React.useState(true);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [nameSearch, setNameSearch] = React.useState('')
+  const [descriptionSearch, setDescriptionSearch] = React.useState('')
+  const [ccssSearch, setCCSSSearch] = React.useState('')
+  const [conceptSearch, setConceptSearch] = React.useState('')
+  const [activityPacksSearch, setActivityPacksSearch] = React.useState('')
+  const [flagSearch, setFlagSearch] = React.useState(DEFAULT_FLAG)
+  const [toolSearch, setToolSearch] = React.useState(DEFAULT_TOOL)
+  const [readabilitySearch, setReadabilitySearch] = React.useState(DEFAULT_READABILITY)
+  const [readabilityOptions, setReadabilityOptions] = React.useState([])
+
+  React.useEffect(() => {
+    if (loading) { getActivities() }
+  }, []);
+
+  function getActivities() {
+    request.get({
+      url: ACTIVITIES_URL,
+    }, (e, r, body) => {
+      if (e || r.statusCode !== 200) {
+        setLoading(false)
+      } else {
+        const data = JSON.parse(body);
+        setLoading(false)
+        setActivities(data.activities);
+
+        const readArr = data.activities.map(act => act.readability_grade_level)
+        const readUnique = Array.from(new Set(readArr))
+        const readFiltered = readUnique.filter(c => c !== '' && c !== null).sort()
+        readFiltered.push(DEFAULT_READABILITY)
+        setReadabilityOptions(readFiltered)
+      }
+    })
+  }
+
+  function getFilteredActivities() {
+    return activities.filter(act => {
+      return (
+        (nameSearch === '' || (act.name && act.name.toLowerCase().includes(nameSearch.toLowerCase()))) &&
+        (descriptionSearch === '' || (act.description && act.description.toLowerCase().includes(descriptionSearch.toLowerCase()))) &&
+        (ccssSearch === '' || (act.standard && act.standard.name.toLowerCase().includes(ccssSearch.toLowerCase()))) &&
+        (conceptSearch === '' || (act.activity_category && act.activity_category.name.toLowerCase().includes(conceptSearch.toLowerCase()))) &&
+        (activityPacksSearch === '' || (act.unit_template_names && act.unit_template_names.some(ut => ut.toLowerCase().includes(activityPacksSearch.toLowerCase())))) &&
+        (flagSearch === DEFAULT_FLAG || (act.data && act.data['flag'] && act.data['flag'] === flagSearch)) &&
+        (toolSearch === DEFAULT_TOOL || (act.classification && act.classification.key === toolSearch)) &&
+        (readabilitySearch === DEFAULT_READABILITY || (act.readability_grade_level && act.readability_grade_level === readabilitySearch)) &&
+        (selectedActivities.length === 0 || !selectedActivities.map(a => a.id).includes(act.id))
```

Comment:
This is a pretty enormous string of conditionals. I think there are a couple of options for refactors here, but the simplest one would be to have each attribute get its own constant (along the lines of `const nameMatch = nameSearch === '' || (act.name && act.name.toLowerCase().includes(nameSearch.toLowerCase())`) and then just check that all the constants are true in the final return statement.